org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/parser/XMLScanner.java

```diff
@@ -272,18 +272,26 @@ TokenType internalScan() {
 				return finishToken(offset, TokenType.PrologEnd);
 			}
 			
-				lastAttributeName = nextAttributeName();
-				if (lastAttributeName.length() > 0) {
-					state = ScannerState.AfterAttributeName;
-					return finishToken(offset, TokenType.AttributeName);
-				}
-			
+			lastAttributeName = nextAttributeName();
+			if (lastAttributeName.length() > 0) {
+				state = ScannerState.AfterAttributeName;
+				return finishToken(offset, TokenType.AttributeName);
+			}
+
 			if (stream.advanceIfChar(_FSL)) { // /
-				state = ScannerState.WithinContent;
+				state = ScannerState.WithinTag;
 				if(stream.advanceIfChar(_RAN)) { // >
+					state = ScannerState.WithinContent;
 					return finishToken(offset, TokenType.StartTagSelfClose);
 				}
+				return finishToken(offset, TokenType.Unknown);
+			}
+		
+			if (stream.peekChar() == _DQO || stream.peekChar() == _SIQ) { // " || '

```

Comment:
maybe store the stream.peekChar() value to avoid all the underlying computations calling peekChar() everytime